apriltag_ros/src/common_functions.cpp

```diff
@@ -222,21 +235,44 @@ AprilTagDetectionArray TagDetector::detectTags (
                                   .buf = gray_image.data
   };
 
-  image_geometry::PinholeCameraModel camera_model;
-  camera_model.fromCameraInfo(camera_info);
+  // publish the header once for the tracker to know this detector exists
+  if (optical_frame_id_ != camera_info->header.frame_id)
+  {
+    optical_frame_id_ = camera_info->header.frame_id;
+    std_msgs::String optical_frame_id_msg;
+    optical_frame_id_msg.data = optical_frame_id_;
+    optical_frame_pub_.publish(optical_frame_id_msg);
```

Comment:
why does the tracker need to know this? doesnt it come through on the header of the tag array message?